### PR TITLE
Add some more definitions

### PR DESCRIPTION
**Pull request by crumblingstatue**
As needed by [my FFI bindings to libnotify](https://github.com/crumblingstatue/rust-libnotify-sys).

### CODE REVIEW

File: src/glib/dataset.rs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#[repr(C)]
+struct _GData;
+
+pub type GData = _GData;
```
Comment (mzabaluev) [inline src/glib/dataset.rs:4]: Why does there need to be `_GData`? It's not a public type, and the generated bindings will just create `struct GData`.

Comment (crumblingstatue) [inline src/glib/dataset.rs:4]: I'll revise this.

File: src/lib.rs [hunk 2]
```diff
@@ -25,6 +25,8 @@
 extern crate libc;
 
 pub mod types;
+pub mod object;
```
Comment (mzabaluev) [inline src/lib.rs:28]: This is a -sys crate dedicated to libglib-2.0, as per [the Cargo convention](http://doc.crates.io/build-script.html#*-sys-packages). Bindings to libgobject should to go [gobject-sys](https://github.com/gi-rust/gobject-sys).

Accordingly, there is no need for glib namespace: all definitions are made at the top level, except the 'intrinsic' types which need to be glob-imported into other crates.

File: src/object/gtype.rs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+use types::gsize;
+
+pub type GType = gsize;
```
Comment (mzabaluev) [inline src/object/gtype.rs:3]: All types in this module are, in fact, already defined in gobject-sys (though some of them may lack fields that are unimportant to Rust bindings).

**Comment by mzabaluev** [comment]
The general intent is to have the -sys crates automatically generated from GIR: gi-rust/grust#24.
So glib-sys and gobject-sys should be kept as close to the eventual output of the generator in the "-sys" mode, using only the information on C API that is available from GIR files.

**Comment by mzabaluev** [comment]
Thanks for the effort @crumblingstatue! I'm looking forward to additions that will keep future compatibility with the (sadly undocumented) generated API.

**Comment by crumblingstatue** [comment]
Thanks for telling me about gobject-sys. I'm going to close this pull request, because these definitions are contained in gobject-sys.

**Pull request closed without merging**